### DISCUSSION TITLE
Postal x Spamassassin - Incoming emails not flagged as spam

### DISCUSSION THREAD

**Discussion by aepau** [postalserver/postal#2358] [Help with using Postal]
Hello team,

I'd like some help about my usage of Postal and Spamassassin. I have a working Postal (upgraded from v1 recently), which I added spamassassin to detect some domains as spam.
I configured Postal as described in the documentation, and my spamassassin configuration seems good because he succeed to detect the mail as spam. See log:

```
Fri Feb 17 09:35:18 2023 [1162] info: spamd: connection from 127.0.0.1 [127.0.0.1]:45238 to port 783, fd 6
Fri Feb 17 09:35:18 2023 [1162] info: spamd: checking message <XXX@mail.gmail.com> for (unknown):998
Fri Feb 17 09:35:18 2023 [1162] info: spamd: identified spam (94.0/5.0) for (unknown):998 in 0.2 seconds, 3270 bytes.
```

Checking the mail in Postal UI doesn't add any Spam flags, nor change the spam score. In Spam Checks Panel for the message, "*This message doesn't have any spam checks. This likely means we haven't scanned this message to determine its likelyhood of being spam. It may take a few seconds to appear after a new message is received.*" which is strange because it actually did it...

 Can you please tell me what could I be missing? What are the prior checks to make when this happens?

**Comment by willpower232** [comment]
By default, Postal does not check outgoing messages for spam, only incoming. Is it possible you're looking at an outbound message rather than one Postal has received?

If you want to check outgoing messages then you can enter a number in this advanced setting

![image](https://user-images.githubusercontent.com/1619102/219943715-9cf32526-f759-48c9-8e59-ec591e0e1cf1.png)

**Reply by aepau** [reply]
Hey @willpower232 thanks for your reply. That's actually an incoming email :sweat_smile:

**Reply by aepau** [reply]
In order to help, I figured out that rspamd message inspector was implemented, so I decided to give it a try. Without any more configuration in postal.yml and blacklist configuration in rspamd, it works!

**Reply by uhlhosting** [reply]
@aepau how you deployed rspamd? Docker or locally?

**Reply by aepau** [reply]
@uhlhosting I did it locally, but I guess both works if you expose the right ports.
